src/components/Modal/src/ModalLayer.vue

```diff
@@ -86,14 +88,20 @@ const apiMixin = {
 			async close() {
 				const isModalActive = !this.state.vnode; // Verify there's no modal on top
 
-				if (isModalActive && vm.currentLayer) {
+				if (isModalActive && vm.currentModalLayer) {
 					if (typeof this.state.options.beforeCloseHook === 'function') {
 						if (!(await this.state.options.beforeCloseHook())) {
 							return; // cancel
 						}
 					}
 
-					vm.currentLayer.state.vnode = undefined; // close modal
+					// Close the open popover (if present) and then close the modal in the next tick.
+					// Closing at the same time will result in the popover content becoming inline and
+					// causes a weird content shift as the modal fades away.
+					vm.popoverApi.closePopover();
```

Comment:
This is likely necessary because we don't have a sophisticated layer manager, but want to call this out as potential future code-smell as we're creating internal dependencies between unrelated components. The popover may not be used ever, but now the modal needs to control closing something that might not exist.

Reply:
@tatems This is throwing an error. I think we need to do `vm.popoverApi?.closePopover();` here